src/main/java/org/spongepowered/common/data/processor/multi/entity/AgeableDataProcessor.java

```diff
@@ -0,0 +1,123 @@
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.multi.entity;
+
+import static org.spongepowered.common.data.util.DataUtil.getData;
+
+import com.google.common.collect.ImmutableMap;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityAgeable;
+import net.minecraft.entity.monster.EntityZombie;
+
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableAgeableData;
+import org.spongepowered.api.data.manipulator.mutable.entity.AgeableData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeAgeableData;
+import org.spongepowered.common.data.processor.common.AbstractEntityDataProcessor;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class AgeableDataProcessor extends AbstractEntityDataProcessor<Entity, AgeableData, ImmutableAgeableData> {
+
+    public AgeableDataProcessor() {
+        super(Entity.class);
+    }
+
+    @Override
+    protected AgeableData createManipulator() {
+        return new SpongeAgeableData(0, true);
+    }
+
+    @Override
+    protected boolean doesDataExist(Entity entity) {
+        return entity instanceof EntityZombie || entity instanceof EntityAgeable;
+    }
+
+    @Override
+    protected boolean set(Entity entityCast, Map<Key<?>, Object> keyValues) {
+        Integer age = (Integer) keyValues.get(Keys.AGE);
+        boolean adult = (Boolean) keyValues.get(Keys.IS_ADULT);
+
+        if (entityCast instanceof EntityAgeable) {
+            EntityAgeable entity = (EntityAgeable) entityCast;
+
+            if (age != null) {
```

Comment:
Still a question to this day... it should never be null, since the retrieval method is returning an integer and not an object.